api/actions/serializers.py

```diff
@@ -315,3 +323,60 @@ def create(self, validated_data):
             )
 
         return target.actions.last()
+
+
+class SchemaResponseActionSerializer(BaseActionSerializer):
+
+    create_payload_schema = create_schema_response_action_payload
+
+    class Meta:
+        type_ = 'schema-response-actions'
+
+    permissions = ser.ChoiceField(choices=permissions.API_CONTRIBUTOR_PERMISSIONS, required=False)
+    visible = ser.BooleanField(default=True, required=False)
+    trigger = ser.ChoiceField(choices=SchemaResponseTriggers.char_field_choices())
+
+    target = TargetRelationshipField(
+        target_class=SchemaResponse,
+        read_only=False,
+        required=True,
+        related_view='schema_responses:schema-response-action-list',
+        related_view_kwargs={'schema_response_id': '<target._id>'},
+        filter_key='target__guids___id',
+    )
+
+    def create(self, validated_data):
+        user = self.context['request'].user
+        trigger = validated_data.get('trigger')
+        target = validated_data.pop('target')
+        comment = validated_data.pop('comment', '')
+        try:
+            if trigger == SchemaResponseTriggers.SUBMIT.db_name:
+                required_approvers = target.parent.get_users_with_perm('admin')
+                target.submit(user=user, comment=comment, required_approvers=required_approvers)
+            elif trigger == SchemaResponseTriggers.APPROVE.db_name:
+                target.approve(user=user, comment=comment)
+            elif trigger == SchemaResponseTriggers.ACCEPT.db_name:
+                target.accept(user=user, comment=comment)
+            elif trigger == SchemaResponseTriggers.ADMIN_REJECT.db_name:
+                target.reject(user=user, comment=comment)
+            elif trigger == SchemaResponseTriggers.MODERATOR_REJECT.db_name:
+                target.reject(user=user, comment=comment)
+            else:
+                raise JSONAPIAttributeException(attribute='trigger', detail='Invalid trigger.')
+        except InvalidTriggerError:
```

Comment:
What would be raising this? Looks like it's only raised in the MachineableMixin. We just raise MachineError on the ApprovalsStateMachine. Worth a TODO to raise the internal error [here](https://github.com/CenterForOpenScience/osf.io/blob/c0da58edb7dadd2f5872cee7f4bfbfb767a4175d/osf/utils/machines.py#L370)?

Also, similarly to below, I would call this a 409, as a MachineError is definitionally a state conflict